src/preprocessing.py

```diff
@@ -53,13 +57,23 @@ def preprocess_data():
             # Process Myelin
             myelin_seg = cv2.imread(myelin_seg_path, cv2.IMREAD_GRAYSCALE)
             myelin_seg_regions = utils.find_regions(myelin_seg)
-            for i, region in enumerate(myelin_seg_regions):
-                minr, minc, maxr, maxc = region.bbox
-                cv2.rectangle(img, (minc, minr), (maxc, maxr), (0, 255, 0), 2)  # Green rectangle for myelin
-                bbox_data.append({"image_name": f"{subject}_{sample}.png", "xmin": minc, "ymin": minr, "xmax": maxr, "ymax": maxc, "class": "myelin"})
+            with open(os.path.join(processed_masks_dir, label_name), "w") as file:
+                for i, region in enumerate(myelin_seg_regions):
+                    minr, minc, maxr, maxc = region.bbox
+                    centroidx, centroidy = region.centroid[0], region.centroid[1]
+                    width,height = region.axis_major_length, region.axis_minor_length
+                    # cv2.rectangle(img, (minc, minr), (maxc, maxr), (0, 255, 0), 2)  # Green rectangle for myelin
+                    bbox_data.append({"image_name": f"{subject}_{sample}.png", "xmin": minc, "ymin": minr, "xmax": maxr, "ymax": maxc, "class": "myelin"})
+
+                    # Normalize coordinates
+                    img_height, img_width = img.shape[:2]
+                    centroidx /= img_width
+                    centroidy /= img_height
+                    width /= img_width
+                    height /= img_height
+                    file.write('0 {} {} {} {}\n'.format(centroidx, centroidy, width, height))
+
 
-            # Save the image with bounding boxes
-            image_name = f"{subject}_{sample}.png"
             cv2.imwrite(os.path.join(processed_images_dir, image_name), img)
```

Comment:
Ok so here, you are writing `img` in `dataset/images/train`. These images have bounding boxes overlaid on them. Is this intentional? These images are great for visual inspection of the data, but if we fed this to the detection model we would effectively be "cheating", wouldn't we?

Reply:
Also corrected